### PR TITLE
feat(provider/kubernetes): v2 reference docs

### PR DESCRIPTION
**Pull request by lwander**
I will add more to the "resource mapping" section in a separate PR to keep this from growing too large.

### CODE REVIEW

**Comment by dorbin** [comment]
@duftler , does this look ok to you? I want to merge it then do an editing run on it in a separate pr.

File: reference/providers/kubernetes-v2/index.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+# The Manifest Based Approach
+
+The Kubernetes Provider V2 combines the strengths of Kubernetes' [declarative
+infrastructure
+management](https://kubernetes.io/docs/tutorials/object-management-kubectl/declarative-object-management-configuration/)
+with Spinnaker's workflow engine for imperative steps when you need them. You
+can fully specify all your infrastructure in the native Kubernetes manifest
+format, while still being able to express, for example, a multi-regional canary
+driven rollout. This represents a significant departure from how deployments
+are managed in Spinnaker today using other provider (including the Kubernetes
```
Comment (duftler) [inline reference/providers/kubernetes-v2/index.md:24]: s/provider/providers

File: reference/providers/kubernetes-v2/index.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+Provider V1), so the below sub-sections will attempt to
+explain the differences.
+
+## No Restrictive Naming Policies
+
+In order to allow users to deploy existing manifests without rewriting them to
+adhere to the [Frigga naming convention](https://github.com/Netflix/frigga),
+resources relationships (between applications/clusters) are managed
```
Comment (duftler) [inline reference/providers/kubernetes-v2/index.md:32]: s/resources/resource

Comment (duftler) [inline reference/providers/kubernetes-v2/index.md:32]: e.g. before 'between'?

File: reference/providers/kubernetes-v2/index.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+See the [Kubernetes API
+conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#spec-and-status)
+for a description of an edge- vs. level-based API.
+
+Prior providers in Spinnaker focused on tracking an operation that modified a
+cloud resource. For example, if a resize operation was run, Spinnaker
+would monitor that operation until the specific resize target was met. This
+runs counter to the Kubernetes level-based API, which only cares about the
+desired state of a resource being satisfied. In order to address this, we
+introduced the concept of __Manifest Stability__:
```
Comment (duftler) [inline reference/providers/kubernetes-v2/index.md:51]: introduced or introduce?

File: reference/providers/kubernetes-v2/index.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+A deployed manifest is consided __stable__ when the Kubernetes
+Controller-Manager no longer needs to modify it, and it is "ready". This
+definition is clearly very dependent on the manifest's kind (e.g. `Service` vs.
+`StatefulSet`), but for example, a `Deployment` is stable when its managed pods
+are updated, available, and ready (meaning they are running your desired
+container, and are serving traffic). Each kind's stability requirements are
```
Comment (duftler) [inline reference/providers/kubernetes-v2/index.md:58]: containers?

File: reference/providers/kubernetes-v2/index.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+detailed in the [Resource Mapping](#resource-mapping) section.
+
+This concept of Manifest Stability is how Spinnaker ensures that operations
+have succeeded. Since there are a number of reasons why a manifest never
```
Comment (duftler) [inline reference/providers/kubernetes-v2/index.md:62]: s/never becomes stables/may never become stable

File: reference/providers/kubernetes-v2/index.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+## Deploying Manifests Stored Externally to Spinnaker
+
+To allow users to store & version their manifest definitions in Git (or
+elsewhere) outside of Spinnaker's pipeline store, we needed another way to
+supply manifest definitions to Pipeline executions.
+
+Using Spinnaker's new Artifact mechanism, we allow file modifications/creations
+to be surfaced as artifacts in pipeline executions. For example, you can now
+configure a pipeline that triggers either whenever 1. a new Docker image is
```
Comment (duftler) [inline reference/providers/kubernetes-v2/index.md:76]: The numbered list looks a little weird rendered within the paragraph.

File: reference/providers/kubernetes-v2/index.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+uploaded, or 2. your manifest file is changed in git.
+
+# Reserved Annotations
```
Comment (duftler) [inline reference/providers/kubernetes-v2/index.md:79]: I know you mentioned this concept up above, but could still use an intro sentence or two here.

File: reference/providers/kubernetes-v2/index.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+
+  The cluster this resource belongs to. This is purely a logical grouping for
+  rendering resources in the UI, and to help with dynamic target selection in
+  Pipeline stages. For example, some stages allow you to select "the newest
+  workload in cluster __X__". How you set up these groupings depends on your
+  delivery needs.
+
+* `moniker.spinnaker.io/stack`, and `moniker.spinnaker.io/detail`
+
+  Stack and detail simply provide ways to group resources using Spinnaker's
```
Comment (duftler) [inline reference/providers/kubernetes-v2/index.md:97]: Worth mentioning their use in sequential naming as well?

File: reference/providers/kubernetes-v2/index.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+  cluster filters as well as apply policies such as [Traffic
+  Guards](https://blog.spinnaker.io/can-i-push-that-building-safer-low-risk-deployments-with-spinnaker-a27290847ac4).
+
+# Resource Mapping
+
+The resource mapping between Spinnaker and Kubernetes constructs is a lot more
```
Comment (duftler) [inline reference/providers/kubernetes-v2/index.md:103]: s/is a lot/is by necessity a lot

**Review by duftler — APPROVED** [review]
LGTM.

**Pull request merged**